File: GrowthWare_CS/GrowthWareFramework/ModelObjects/MStateProfile.cs

using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Web;
using GrowthWare.Framework.ModelObjects.Base;
using GrowthWare.Framework.Enumerations;

namespace GrowthWare.Framework.ModelObjects
{
	[Serializable(), CLSCompliant(true)]
	public class MStateProfile : MProfile
	{

		#region "Private Methods"
		private new void init(DataRow Datarow)
		{
			base.init(Datarow);
			base.setString(ref mState, ref Datarow, "STATE");
			base.setString(ref mDescription, ref Datarow, "DESCRIPTION");
			base.setInt(ref mSTATUS_SEQ_ID, ref Datarow, "STATUS_SEQ_ID");
		}
		#endregion

		#region "Private Properties"
		    private string mState = "NEG1";
		    private string mDescription = string.Empty;
            private int mSTATUS_SEQ_ID = (int)SystemStatus.Inactive;
        #endregion

		#region "Public Properties"
		public string State {
			get { return mState; }
			set { mState = value.Trim(); }
		}

		public string Description {
			get { return mDescription; }
			set { mDescription = value.Trim(); }
		}

		public int STATUS_SEQ_ID {
			get { return mSTATUS_SEQ_ID; }
			set { mSTATUS_SEQ_ID = value; }
		}
		#endregion

		#region "Public Methods"
		/// <summary>
		/// Will return an instance populated with default values.
		/// </summary>
		/// <remarks></remarks>

		public MStateProfile()
		{
		}

		/// <summary>
		/// Will return an instance populated with information for the data row provided.
		/// </summary>
		/// <param name="dr"></param>
		/// <remarks></remarks>
		public MStateProfile(DataRow dr)
		{
			init(dr);
		}
		#endregion

	}
}

File: GrowthWare_VB/GrowthWareFrameworkModel/Profiles/MStateProfile.vb
Imports GrowthWare.Framework.ModelObjects.Base
Imports GrowthWare.Framework.Enumerations

Namespace ModelObjects
	<Serializable(), CLSCompliant(True)> _
	Public Class MStateProfile
		Inherits MProfile

#Region "Private Methods"
		Private Shadows Sub Init(ByVal dr As DataRow)
			MyBase.Init(dr)
			mState = dr("STATE")
			mDescription = dr("DESCRIPTION")
			mSTATUS_SEQ_ID = dr("STATUS_SEQ_ID")
		End Sub
#End Region

#Region "Private Properties"
		Private mState As String = "NEG1"
		Private mDescription As String = String.Empty
		Private mSTATUS_SEQ_ID As Integer = SystemStatus.Inactive
#End Region

#Region "Public Properties"
		Public Property State() As String
			Get
				Return mState
			End Get
			Set(ByVal value As String)
				mState = value.Trim
			End Set
		End Property

		Public Property Description() As String
			Get
				Return mDescription
			End Get
			Set(ByVal value As String)
				mDescription = value.Trim
			End Set
		End Property

		Public Property STATUS_SEQ_ID() As Integer
			Get
				Return mSTATUS_SEQ_ID
			End Get
			Set(ByVal value As Integer)
				mSTATUS_SEQ_ID = value
			End Set
		End Property
#End Region

#Region "Public Methods"
		''' <summary>
		''' Will return an instance populated with default values.
		''' </summary>
		''' <remarks></remarks>
		Public Sub New()

		End Sub

		''' <summary>
		''' Will return an instance populated with information for the data row provided.
		''' </summary>
		''' <param name="dr"></param>
		''' <remarks></remarks>
		Public Sub New(ByVal dr As DataRow)
			Init(dr)
		End Sub
#End Region

	End Class
End Namespace